src/bf_functions.py

```diff
@@ -344,231 +291,184 @@ def frequency2fft_bin(self, frequencies=all):
+        # if optimisations are enabled and (same data, and multiple targets)
+        # write to all targets with the same register name
         if self.optimisations and (len(data) == 1 and len(targets) > 1):
-            if self.config.simulate == True: print 'optimisations on, single data item, writing in parallel'
-
-            datum = data[0]
-            n_bfs = len(self.get_bfs())
-
-            #run through all bfs
-            for bf_index in range(n_bfs):
-                fpgas = []
-
-                #generate register name
-                name = '%s%s_%s' %(bf_register_prefix, bf_index, device_name)
-
-		datum = struct.pack(">I", data[0])
-
-                #find all fpgas to be written to for this bf
-                for target in targets:
-                    if target['bf'] == bf_index:
-                        fpgas.append(target['fpga'])
-
-                if len(fpgas) != 0:
-                    if self.config.simulate == True: print 'dummy executing non-blocking request for write to %s on %d fpgas' %(name, len(fpgas))
-                    else:
-
-                        nottimedout, rv = corr.corr_functions.non_blocking_request(fpgas = fpgas, \
-                                                                       		   timeout = timeout, \
-                                                                                   request = 'write', \
-                                                                                   request_args = [name, offset*4, datum])
-
-                        if nottimedout == False:
-			    raise fbfException(1, 'Timeout asynchronously writing 0x%.8x to %s on %d fpgas offset %i' %(data[0], name, len(fpgas), offset), \
-			                       'function %s, line no %s\n' %(__name__, inspect.currentframe().f_lineno), \
-					       self.syslogger)
-                        for k, v in rv.items():
-                            if v['reply'] != 'ok':
-				raise fbfException(1, 'Got %s instead of ''ok'' when writing 0x%.8x to %s:%s offset %i' %(v['reply'], data[0], k, name, offset), \
-						   'function %s, line no %s\n' %(__name__, inspect.currentframe().f_lineno), \
-						   self.syslogger)
-
-        #optimisations off, or (many data, or a single target) so many separate writes
-        else:
-            if self.config.simulate == True: print 'optimisations off or multiple items or single target'
-
-            for target_index,target in enumerate(targets):
-
-                if len(data) == 1 and len(targets) > 1: datum = data[0]
-                else: datum = data[target_index]
-
-		if datum < 0: datum_str = struct.pack(">i", datum)
-		else: datum_str = struct.pack(">I", datum)
-
-                name = '%s%s_%s' %(bf_register_prefix, target['bf'], device_name)
-
-                #pretend to write if no FPGA
+            if self.config.simulate == True:
+                print 'optimisations on, single data item, writing in parallel'
+        datum = data[0]
```

Comment:
The meaning here is changed, this block would previously only have run when both if statements above were true, will now always run.